operator/src/main/java/oracle/kubernetes/operator/rest/RestBackendImpl.java

```diff
@@ -89,13 +97,38 @@
   RestBackendImpl(String principal, String accessToken, Collection<String> domainNamespaces) {
     LOGGER.entering(principal, domainNamespaces);
     this.principal = principal;
-    userInfo = authenticate(accessToken);
+    initializeCallBuilder(accessToken);
     this.domainNamespaces = domainNamespaces;
     LOGGER.exiting();
   }
 
+  private void initializeCallBuilder(String accessToken) {
```

Comment:
It would probably be cleaner to add a fluent method to CallBuilder that would allow you to say something like:
```
  new CallBuilder().withAuthentication(new AccessTokenAuthentication(accessToken))
```

If the authentication is null, you'd get the current behavior. That would allow this logic to be moved into CallBuilder, where it really belongs. This isn't a REST thing.

But this whole bit of code seems needlessly complex. Find a way to simplify it, please.

Reply:
moved AccessTokenAuthentication logic CallBuilder as suggested.  Isolated to callbuilder as that's where it's more suited and relevant.